libraries/standard/http/src/http_client.c

```diff
@@ -3,59 +3,75 @@
 
 #include "http_client.h"
 #include "private/http_client_internal.h"
-#include "private/http_client_parse.h"
-#include "http_parser/http_parser.h"
 
 /*-----------------------------------------------------------*/
 
 /**
- * @brief An aggregator that represents the user-provided parameters to the
- * #HTTPClient_ReadHeader API function. This will be used as context parameter
- * for the parsing callbacks used by the API function.
+ * @brief Send HTTP bytes over the transport send interface.
+ *
+ * @param[in] pTransport Transport interface.
+ * @param[in] pBuffer HTTP request data to send.
+ * @param[in] bufferLen HTTP request data length.
+ *
+ * @return #HTTP_SUCCESS if successful. If there was a network error or less
+ * bytes than what were specified were sent, then #HTTP_NETWORK_ERROR is
+ * returned.
  */
-typedef struct findHeaderContext
-{
-    const uint8_t * pField;
-    size_t fieldLen;
-    const uint8_t ** pValueLoc;
-    size_t * pValueLen;
-    uint8_t fieldFound : 1;
-    uint8_t valueFound : 1;
-} findHeaderContext_t;
-
-/*-----------------------------------------------------------*/
+static HTTPStatus_t sendHttpData( const HTTPTransportInterface_t * pTransport,
+                                  const uint8_t * pBuffer,
+                                  size_t bufferLen );
 
 /**
  * @brief Send the HTTP headers over the transport send interface.
  *
  * @param[in] pTransport Transport interface.
  * @param[in] pRequestHeaders Request headers to send, it includes the buffer
  * and length.
+ * @param[in] reqBodyBufLen The length of the request body to be sent. This is
```

Comment:
Is it the request body buffer length or the actual request body length?

Reply:
It is the length of the request body.